Source/Objects/Custom Hardware/SNO+/XL3/ORXL3Model.m

```
@@ -4114,6 +4114,39 @@ - (void) setHVRelays:(unsigned long long)aRelayMask
     }
 }
 
+- (void) readHVRelays:(uint64_t*) _relayMask isKnown:(BOOL*)isKnown {
+    char payload[XL3_PAYLOAD_SIZE];
+    uint32_t mask1, mask2, known;
+
+    memset(payload, 0, XL3_PAYLOAD_SIZE);
+    GetHVRelaysResults* data = (GetHVRelaysResults*) payload;
+
+    @try {
+        [[self xl3Link] sendCommand:GET_HV_RELAYS_ID withPayload:payload expectResponse:YES];
+    }
+    @catch (NSException *exception) {
+        NSLogColor([NSColor redColor],@"%@ error sending GetHVRelays command.\n",[[self xl3Link] crateName]);
```

Should print the exception name and reason here. Something like:

```
NSLogColor([NSColor redColor], @"crate %02d: error getting HV relay state. error: %@ reason: %@", [self crateNumber], [e name], [e reason]);
```

If you are going to let the exception bubble up though, it's probably just best to not wrap this in a try except block.